oaipmh/requests/data_queries.py

```diff
@@ -0,0 +1,143 @@
+from typing import Dict, Union
+import re
+from datetime import datetime, timezone
+
+from arxiv.taxonomy.definitions import GROUPS, ARCHIVES_ACTIVE, CATEGORIES_ACTIVE
+from arxiv.taxonomy.category import Group, Archive, Category
+
+from oaipmh.data.oai_config import SUPPORTED_METADATA_FORMATS, EARLIEST_DATE
+from oaipmh.data.oai_errors import OAIBadArgument, OAIBadFormat
+from oaipmh.data.oai_properties import OAIParams, OAIVerbs
+from oaipmh.serializers.output_formats import Response
+from oaipmh.requests.param_processing import process_identifier
+
+DATE_REGEX = r"\d{4}-\d{2}-\d{2}"
+
+def get_record(params: Dict[str, str]) -> Response:
+    """used to get data on a particular record in a particular metadata format"""
+    query_data: Dict[OAIParams, str]={OAIParams.VERB:OAIVerbs.GET_RECORD}
+
+    # get parameters
+    expected_params={OAIParams.ID, OAIParams.META_PREFIX, OAIParams.VERB}
+    if set(params.keys()) != expected_params:
+        raise OAIBadArgument(f"Parameters provided did not match expected. Expected: {', '.join(str(param) for param in expected_params)}")
+    
+    identifier_str=params[OAIParams.ID]
+    arxiv_id=process_identifier(identifier_str)
+    query_data[OAIParams.ID]=identifier_str
+
+    meta_type_str=params[OAIParams.META_PREFIX]
+    if meta_type_str not in SUPPORTED_METADATA_FORMATS:
+        raise OAIBadFormat(reason="Did not recognize requested format", query_params=query_data)
+    meta_type=SUPPORTED_METADATA_FORMATS[meta_type_str]
+
+    #TODO paramters done, do rest of function
+
+    return "<a>b</a>", 200, {}
+
+def list_records(params: Dict[str, str]) -> Response:
```

Comment:
list_records and list_identifiers may be unnecessary extra level of function calls.